root/static/scripts/account/components/ApplicationForm.js

```diff
@@ -57,18 +57,19 @@ class ApplicationForm extends React.Component<Props, State> {
 
   handleOauthRedirectURIChange(e: SyntheticEvent<HTMLInputElement>) {
     const selectedOauthRedirectURI = e.currentTarget.value;
-    this.setState(prevState => mutate<State, _>(prevState, newState => {
-      newState.form.field.oauth_redirect_uri.value = selectedOauthRedirectURI;
-    }));
+    this.setState(prevState => mutate(prevState)
+      .set('form', 'field', 'oauth_redirect_uri', 'value',
+           selectedOauthRedirectURI)
```

Comment:
BTW, the property names and values are still type-checked: so if you make a typo in one of the keys, or pass the wrong type of value at the end, Flow will error.